### PR TITLE
Feature/typescript integration

### PR DESCRIPTION
**Pull request by aungkhantkyaw9**
## Description

Add typescript eslint to eslint config rules

## Tasks
- [x] Install typescript
- [x] Setup typescript for eslint

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -159,5 +159,10 @@ module.exports = {
                 'classPropertiesAllowed': false
             }
         ]
-    }
+    },
+    'overrides': [{
+        // for files matching this pattern
```
Comment (dmoskovtsov) [inline index.js]: - Please remove the comment. 
- Please setup a webstorm to use single quotes & reformat this file.

File: package.json [hunk 2]
```diff
@@ -17,25 +17,28 @@
...
     "eslint-plugin-no-else": "0.2.2",
-    "eslint-plugin-sonarjs": "0.15.0",
-    "eslint-plugin-prefer-arrow": "1.2.3"
+    "eslint-plugin-prefer-arrow": "1.2.3",
+    "eslint-plugin-react": "7.31.10",
+    "eslint-plugin-sonarjs": "0.15.0"
   },
   "dependencies": {
+    "@typescript-eslint/eslint-plugin": "^5.54.0",
+    "@typescript-eslint/parser": "^5.54.0",
```
Comment (dmoskovtsov) [inline package.json]: Use exact versions (unless you build a library). 

   "@typescript-eslint/eslint-plugin": "5.54.0",

File: package.json [hunk 3]
```diff
@@ -17,25 +17,28 @@
...
     "eslint-plugin-folders": "1.0.3",
-    "eslint-plugin-react": "7.31.10",
-    "eslint-plugin-detect-bad-words": "1.0.7",
     "eslint-plugin-mocha": "10.1.0",
     "eslint-plugin-no-else": "0.2.2",
+    "eslint-plugin-prefer-arrow": "1.2.3",
+    "eslint-plugin-react": "7.31.10",
     "eslint-plugin-sonarjs": "0.15.0",
-    "eslint-plugin-prefer-arrow": "1.2.3"
+    "typescript": "^4.9.5"
```
Comment (dmoskovtsov) [inline package.json]: Same here

**Review by dmoskovtsov — CHANGES_REQUESTED** [review]
- note, you might have to put new dependencies to peer dependencies.

**Comment by aungkhantkyaw9** [comment]
@dmoskovtsov after testing clean install of yarn in `store` , we don't need to move typescript related packages from `dependencies` to `peer-dependencies`. All necessary packages are automatically installed in `store`

**Review by dmoskovtsov — APPROVED** [review]

**Pull request merged**